pkg/kubelet/kubelet_node_status.go

```diff
@@ -554,3 +571,52 @@ func validateNodeIP(nodeIP net.IP) error {
 	}
 	return fmt.Errorf("Node IP: %q not found in the host's network interfaces", nodeIP.String())
 }
+
+// nodeStatusHasChanged compares the original node and current node's status and
+// returns true if any change happens. The heartbeat timestamp is ignored.
+func nodeStatusHasChanged(originalStatus *v1.NodeStatus, status *v1.NodeStatus) bool {
+	if originalStatus == nil || status == nil {
+		return !apiequality.Semantic.DeepEqual(originalStatus, status)
+	}
+
+	if nodeConditionsHaveChanged(originalStatus.Conditions, status.Conditions) {
+		return true
+	}
+
+	originalConditions := originalStatus.Conditions
+	originalStatus.Conditions = nil
+	conditions := status.Conditions
+	status.Conditions = nil
+	defer func() {
+		originalStatus.Conditions = originalConditions
+		status.Conditions = conditions
+	}()
+	return !apiequality.Semantic.DeepEqual(originalStatus, status)
+}
+
+// nodeConditionsHaveChanged compares the original node and current node's
+// conditions and returns true if any change happens. The heartbeat timestamp is
+// ignored.
+func nodeConditionsHaveChanged(originalConditions []v1.NodeCondition, conditions []v1.NodeCondition) bool {
+	if len(originalConditions) != len(conditions) {
+		return true
+	}
+
+	originalConditionsCopy := make([]v1.NodeCondition, len(originalConditions))
```

Comment:
`make([]v1.NodeCondition, 0, len(originalConditions))` does not work because the length is 0. The `copy()` below only works for min of `len(originalConditionsCopy)` and `len(originalConditions)`, which is then 0. And I think allocating 0 values should be fast, right?

Reply:
It's allocating the whole struct (it's not that much but it's something). But yeah - you have good point about copy.

You may also try the following - that shouldn't allocate even those empty values.
```
originalConditionsCopy := make([]v1.NodeCondition, 0, len(originalConditions))
originalConditionsCopy = append(originalConditionsCopy, originalConditions)
```

[But yeah - it's not extremely important]

Reply:
Done